### PR TITLE
Relics Instant Cast Finders

### PR DESCRIPTION
**Pull request by Linkaaaaa**
Tracked:

- Vass
- Firebrand
- Isgarren
- Peitha
- Dragonhunter
- Cerus
- Ice
- Fireworks
- Citadel
- Nightmare
- Krait
- Wizard's Tower

Also added some more icons and renamed some relics that were showing the number or UNKNOWN in the html.

Zip with the test logs:
[relics.zip](https://github.com/baaron4/GW2-Elite-Insights-Parser/files/12448140/relics.zip)

### CODE REVIEW

File: GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs [hunk 1]
```diff
@@ -25,6 +25,46 @@ internal static class ProfHelper
             new EffectCastFinderByDst(RuneOfNightmare, EffectGUIDs.RuneOfNightmare).UsingOrigin(InstantCastFinder.InstantCastOrigin.Gear),
             new BuffGainCastFinder(PortalEntranceWhiteMantleWatchwork, PortalWeavingWhiteMantleWatchwork),
             new BuffGainCastFinder(PortalExitWhiteMantleWatchwork, PortalUsesWhiteMantleWatchwork).UsingBeforeWeaponSwap(true),
+            // Relics
+            new BuffGainCastFinder(RelicOfVass, RelicOfVass).UsingOrigin(InstantCastFinder.InstantCastOrigin.Gear),
+            new BuffGainCastFinder(RelicOfTheFirebrand, RelicOfTheFirebrand).UsingOrigin(InstantCastFinder.InstantCastOrigin.Gear),
+            new BuffGainCastFinder(RelicOfIsgarrenTargetBuff, RelicTargetPlayerBuff).UsingChecker((bae, combatData, agentData, skillData) =>
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:31]: The RelicTargetPlayerBuff is unreliable, it does not refresh as the RelicOfXXXTargetBuff gets refreshed on the target.

File: GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs [hunk 2]
```diff
@@ -25,6 +25,46 @@ internal static class ProfHelper
...
+                return combatData.GetBuffData(RelicOfPeithaTargetBuff).Where(x => x.CreditedBy == bae.To && Math.Abs(x.Time - bae.Time) < ServerDelayConstant).Any();
+            }).UsingOrigin(InstantCastFinder.InstantCastOrigin.Gear),
+
+            new BuffGainCastFinder(RelicOfTheDragonhunterTargetBuff, RelicTargetPlayerBuff).UsingChecker((bae, combatData, agentData, skillData) =>
+            {
+                return combatData.GetBuffData(RelicOfTheDragonhunterTargetBuff).Where(x => x.CreditedBy == bae.To && Math.Abs(x.Time - bae.Time) < ServerDelayConstant).Any();
+            }).UsingOrigin(InstantCastFinder.InstantCastOrigin.Gear),
+            new EffectCastFinder(RelicOfCerusHit, EffectGUIDs.RelicOfCerusEye).UsingOrigin(InstantCastFinder.InstantCastOrigin.Gear),
+            new EffectCastFinder(RelicOfTheIce, EffectGUIDs.RelicOfIce).UsingICD(1000).UsingOrigin(InstantCastFinder.InstantCastOrigin.Gear),
+            new EffectCastFinder(RelicOfFireworks, EffectGUIDs.RelicOfFireworks).UsingOrigin(InstantCastFinder.InstantCastOrigin.Gear),
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:47]: Should remain commented for now, it does not trigger on refresh.

File: GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs [hunk 3]
```diff
@@ -25,6 +25,46 @@ internal static class ProfHelper
...
+            new EffectCastFinder(RelicOfTheCitadel, EffectGUIDs.RelicWhiteCircle).UsingChecker((evt, combatData, agentData, skillData) =>
+            {
+                combatData.TryGetEffectEventsByGUID(EffectGUIDs.RelicOfTheCitadelExplosion, out IReadOnlyList<EffectEvent> effects);
+                return effects != null && effects.Any(x => x.Time > evt.Time);
+            }).UsingOrigin(InstantCastFinder.InstantCastOrigin.Gear),
+            new EffectCastFinder(RelicOfTheNightmare, EffectGUIDs.RelicWhiteCircle).UsingChecker((evt, combatData, agentData, skillData) =>
+            {
+                combatData.TryGetEffectEventsByGUID(EffectGUIDs.RelicOfTheNightmare, out IReadOnlyList<EffectEvent> effects);
+                return effects != null && effects.Any(x => x.Time > evt.Time);
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:56]: Not sure to get that condition, isn't that too permissive? Not Src/Dst check either, what happens if I'm running a relic triggering WhiteCircle and someone else also does it? Wouldn't it multi trigger, for ex:
I run Nightmare, squad mate run Krait, they trigger Krait at time N (so white + krait effect), I trigger Nightmare at time M such as M < N, wouldn't I also register a Krait activation (I activated white and the condition where Krait effect exist with a higher timestamp exists)

File: GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs (hunk 1 above)
Comment (Linkaaaaa) [reply GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:31]: i see both being repplied each time, am i doing something wrong?

[peitha.zip](https://github.com/baaron4/GW2-Elite-Insights-Parser/files/12448279/peitha.zip)

Comment (EliphasNUIT) [reply GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:31]: It does not get reapplied to you if the target has still the buff running, only on new applications, at least that was the case for dragonhunter relic when we tested it with Zerthox.

Comment (EliphasNUIT) [reply GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:31]: https://discord.com/channels/456611641526845473/1139667440465354752/1143698177447448576

Comment (Linkaaaaa) [reply GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:31]: peitha has a projectile effect so i switched to using that, i'll try to produce what was said for isgarren and dragonhunter

File: GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs (hunk 3 above)
Comment (Linkaaaaa) [reply GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:56]: the only secondary effect that has a src is the wizard tower, all the others don't have one

Comment (EliphasNUIT) [reply GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:51]: I'm guessing the extra effects does not happen instantly?

Comment (Linkaaaaa) [reply GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:51]: no they have different delays since the white circles is there as a warning

Comment (EliphasNUIT) [reply GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:51]: In that case they should be commented until we find a better way to detect them, unfortunately the current implementation collides with each other

**Comment by Linkaaaaa** [comment]
[dragonhunter2.zip](https://github.com/baaron4/GW2-Elite-Insights-Parser/files/12448432/dragonhunter2.zip)
some logs with dh relic too

File: GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs [hunk 4]
```diff
@@ -25,6 +25,42 @@ internal static class ProfHelper
...
+            //    return effects != null && effects.Any(x => x.Time > evt.Time);
+            //}).UsingOrigin(InstantCastFinder.InstantCastOrigin.Gear),
+            //new EffectCastFinder(RelicOfTheKrait, EffectGUIDs.RelicWhiteCircle).UsingChecker((evt, combatData, agentData, skillData) =>
+            //{
+            //    combatData.TryGetEffectEventsByGUID(EffectGUIDs.RelicOfTheKrait, out IReadOnlyList<EffectEvent> effects);
+            //    return effects != null && effects.Any(x => x.Time > evt.Time);
+            //}).UsingOrigin(InstantCastFinder.InstantCastOrigin.Gear),
+            new EffectCastFinder(RelicOfTheWizardsTower, EffectGUIDs.RelicWhiteCircle).UsingChecker((evt, combatData, agentData, skillData) =>
+            {
+                combatData.TryGetEffectEventsByGUID(EffectGUIDs.RelicOfTheWizardsTower, out IReadOnlyList<EffectEvent> effects);
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EIData/ProfHelpers/ProfHelper.cs:61]: what is the delay between the circle appearing and the effect triggering?
"x.Time >= evt.Time" does not feel super safe.
Also checkout UsingSecondaryEffectChecker API on EffectCastFinder.

**Comment by EliphasNUIT** [comment]
Thanks!

**Pull request merged**